extensions/apicurio-registry-avro/runtime/src/main/java/io/quarkus/apicurio/registry/binding/ServiceRegistryBindingConverter.java

```diff
@@ -0,0 +1,105 @@
+package io.quarkus.apicurio.registry.binding;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.logging.Logger;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBinding;
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConfigSource;
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConverter;
+
+public class ServiceRegistryBindingConverter implements ServiceBindingConverter {
+
+    private static Logger LOG = Logger.getLogger(ServiceRegistryBindingConverter.class.getName());
+
+    private static final String INCOMING_PREFIX = "mp.messaging.incoming.";
+    private static final String OUTGOING_PREFIX = "mp.messaging.outgoing.";
+
+    @Override
+    public Optional<ServiceBindingConfigSource> convert(List<ServiceBinding> serviceBindings) {
+        var matchingByType = ServiceBinding.singleMatchingByType("serviceregistry", serviceBindings);
+        Config config = ConfigProvider.getConfig();
+        if (matchingByType.isEmpty()) {
+            return Optional.empty();
+        }
+
+        var binding = matchingByType.get();
+
+        List<String> channels = extractChannels(config);
+
+        Map<String, String> properties = new HashMap<>();
+
+        String registryUrl = binding.getProperties().get("registryUrl");
+        if (registryUrl == null) {
+            registryUrl = binding.getProperties().get("registryurl");
+        }
+        if (registryUrl != null) {
+            properties.put("mp.messaging.connector.smallrye-kafka.apicurio.registry.url", registryUrl);
```

Comment:
I would use "kafka." instead of "mp.messaging.connector.smallrye-kafka."

So, bare clients using Kafka default config will have the apicurio URL configured too.